src/main/java/subway/ui/exceptionHandler/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,38 @@
+package subway.ui.exceptionHandler;
+
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler({IllegalArgumentException.class, IllegalStateException.class})
+    public ResponseEntity<String> handleRuntimeException(final Exception e) {
+        e.printStackTrace();
+        return ResponseEntity.badRequest().body(e.getMessage());
+    }
+
+    @ExceptionHandler({DataAccessException.class, EmptyResultDataAccessException.class})
+    public ResponseEntity<Void> handleSQLException(final Exception e) {
+        e.printStackTrace();
+        return ResponseEntity.badRequest().build();
+    }
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<String> handleMethodArgumentNotValidException(final Exception e) {
+        e.printStackTrace();
+        return ResponseEntity.badRequest().body(e.getMessage());
+    }
+
+    @ExceptionHandler(Exception.class)
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    public void handleAll(final Exception e) {
+        e.printStackTrace();
```

Comment:
예외 로깅이나 기타 로깅을 남길때는 로거를 사용하는 습관을 들여두는 게 좋습니다 :) 
```suggestion
    log.error(e);
```

Reply:
로깅 레벨에도 여러 종류가 있는 것을 알게되었습니다.
400번대 에러는 프로덕션 환경에서 굳이 몰라도 되지만 개발 환경에서는 충분히 필요할 수 있는 정보라 log.debug()를 사용하고 500번대 에러는 log.error()을 통해 에러를 확인할 수 있도록 했습니다.
개발 환경과 프로덕션 환경 로그 출력을 분리하기 위해 logback-spring.xml 파일을 설정하여 개발 환경은 debug레벨의 로그를 출력하도록 했고, 프로덕션 환경은 기본 셋팅인 info레벨 이상의 로그를 출력하도록 했습니다.

Reply:
로깅에 대해 잘 학습해주셨군요 ㅎㅎ